designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/activities/ScenarioActivityRepositoryAuditLogDecorator.scala

```diff
@@ -0,0 +1,108 @@
+package pl.touk.nussknacker.ui.process.repository.activities
+
+import db.util.DBIOActionInstances.DB
+import pl.touk.nussknacker.engine.api.deployment._
+import pl.touk.nussknacker.engine.api.process.{ProcessId, VersionId}
+import pl.touk.nussknacker.ui.api.description.scenarioActivity.Dtos.Legacy
+import pl.touk.nussknacker.ui.db.entity.AttachmentEntityData
+import pl.touk.nussknacker.ui.process.ScenarioActivityAuditLog
+import pl.touk.nussknacker.ui.process.ScenarioAttachmentService.AttachmentToAdd
+import pl.touk.nussknacker.ui.process.repository.activities.ScenarioActivityRepository.ModifyCommentError
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import slick.dbio.DBIOAction
+
+import java.time.Clock
+import scala.concurrent.ExecutionContext
+
+class ScenarioActivityRepositoryAuditLogDecorator(
+    underlying: ScenarioActivityRepository
+)(implicit executionContext: ExecutionContext)
+    extends ScenarioActivityRepository {
+
+  def clock: Clock = underlying.clock
```

Comment:
maybe protected?

Reply:
I think I edited wrong comment, it was about making clock protected, but the decorator needs to have access to the clock